### ISSUE TITLE
Time of training 

### ISSUE THREAD

**Issue by cuicathy** [arimousa/DDAD#8]
Hi, 

Thanks for releasing the code! I am using an RTXA6000 to run the main.py for the "Hazelnut" class of the MVTec dataset. It takes over 10 hours to finish the training. I do not have much experience in training a diffusion model. I wonder is it common to take such a long time to train the model and is it necessary to train so many epochs to achieve competitive performance on the MVTec dataset?

Thank you very much :) !

**Comment by arimousa** [comment]
Hi,
Training diffusion models is expensive, taking up to 10 hours is realistic. Hence, I recently published our checkpoints.: https://drive.google.com/drive/u/0/folders/1FF83llo3a-mN5pJN8-_mw0hL5eZqe9fC.
So you can simply download and use them.

Best,
Arian

**Comment by cuicathy** [comment]
Hi Arian, 

Thanks for your prompt reply. I used the default number of training epoch 3000 in your code. However, it seems that the loss did not converge (attached below). Is it a normal situation? How did you decide when to stop training? 

Thanks, 
Cathy
-------------------------------------------------------------------------------------------------------------

hazelnut
 Num params:  32952707
Epoch 25 | Loss: 1023.5911865234375
Epoch 50 | Loss: 736.711181640625
Epoch 75 | Loss: 732.1900024414062
Epoch 100 | Loss: 382.947021484375
Epoch 125 | Loss: 833.1383056640625
Epoch 150 | Loss: 299.0972900390625
Epoch 175 | Loss: 2387.00927734375
Epoch 200 | Loss: 1050.40087890625
Epoch 225 | Loss: 714.7556762695312
Epoch 250 | Loss: 389.92364501953125
Epoch 275 | Loss: 119.1385726928711
Epoch 300 | Loss: 844.815185546875
Epoch 325 | Loss: 326.74737548828125
Epoch 350 | Loss: 515.1982421875
Epoch 375 | Loss: 367.17388916015625
Epoch 400 | Loss: 117.17292785644531
Epoch 425 | Loss: 239.16549682617188
Epoch 450 | Loss: 397.9999694824219
Epoch 475 | Loss: 322.1954345703125
Epoch 500 | Loss: 416.4974670410156
Epoch 525 | Loss: 1526.881103515625
Epoch 550 | Loss: 2376.6162109375
Epoch 575 | Loss: 584.3609619140625
Epoch 600 | Loss: 302.7601013183594
Epoch 625 | Loss: 1230.7706298828125
Epoch 650 | Loss: 1406.0301513671875
Epoch 675 | Loss: 1771.720947265625
Epoch 700 | Loss: 709.1362915039062
Epoch 725 | Loss: 223.4062957763672
Epoch 750 | Loss: 557.409423828125
Epoch 775 | Loss: 1627.894775390625
Epoch 800 | Loss: 538.3485107421875
Epoch 825 | Loss: 1164.44580078125
Epoch 850 | Loss: 759.3908081054688
Epoch 875 | Loss: 96.91363525390625
Epoch 900 | Loss: 1704.71044921875
Epoch 925 | Loss: 344.38018798828125
Epoch 950 | Loss: 521.3500366210938
Epoch 975 | Loss: 575.2868041992188
Epoch 1000 | Loss: 1818.876953125
Epoch 1025 | Loss: 271.4190979003906
Epoch 1050 | Loss: 489.582275390625
Epoch 1075 | Loss: 683.8013916015625
Epoch 1100 | Loss: 543.7854614257812
Epoch 1125 | Loss: 337.1676025390625
Epoch 1150 | Loss: 380.9627685546875
Epoch 1175 | Loss: 485.7334289550781
Epoch 1200 | Loss: 675.9863891601562
Epoch 1225 | Loss: 325.05615234375
Epoch 1250 | Loss: 598.5099487304688
Epoch 1275 | Loss: 334.1776123046875
Epoch 1300 | Loss: 453.4494934082031
Epoch 1325 | Loss: 282.6418151855469
Epoch 1350 | Loss: 369.94818115234375
Epoch 1375 | Loss: 297.21673583984375
Epoch 1400 | Loss: 184.83689880371094
Epoch 1425 | Loss: 933.3197021484375
Epoch 1450 | Loss: 759.6148681640625
Epoch 1475 | Loss: 276.5460510253906
Epoch 1500 | Loss: 1251.544189453125
Epoch 1525 | Loss: 301.88055419921875
Epoch 1550 | Loss: 337.35858154296875
Epoch 1575 | Loss: 777.5225830078125
Epoch 1600 | Loss: 434.32177734375
Epoch 1625 | Loss: 313.37127685546875
Epoch 1650 | Loss: 304.4764404296875
Epoch 1675 | Loss: 499.295654296875
Epoch 1700 | Loss: 280.7455749511719
Epoch 1725 | Loss: 724.482421875
Epoch 1750 | Loss: 261.9753112792969
Epoch 1775 | Loss: 258.490478515625
Epoch 1800 | Loss: 212.65306091308594
Epoch 1825 | Loss: 576.6920776367188
Epoch 1850 | Loss: 555.6858520507812
Epoch 1875 | Loss: 238.76400756835938
Epoch 1900 | Loss: 224.8247528076172
Epoch 1925 | Loss: 340.319580078125
Epoch 1950 | Loss: 760.9388427734375
Epoch 1975 | Loss: 513.8687133789062
Epoch 2000 | Loss: 473.607421875
Epoch 2025 | Loss: 1325.25537109375
Epoch 2050 | Loss: 487.84564208984375
Epoch 2075 | Loss: 836.0723266601562
Epoch 2100 | Loss: 356.18023681640625
Epoch 2125 | Loss: 210.0155029296875
Epoch 2150 | Loss: 438.1662292480469
Epoch 2175 | Loss: 413.62493896484375
Epoch 2200 | Loss: 782.3084716796875
Epoch 2225 | Loss: 332.5396423339844
Epoch 2250 | Loss: 1018.5026245117188
Epoch 2275 | Loss: 337.9665832519531
Epoch 2300 | Loss: 522.52880859375
Epoch 2325 | Loss: 322.390625
Epoch 2350 | Loss: 312.58221435546875
Epoch 2375 | Loss: 214.66934204101562
Epoch 2400 | Loss: 460.0513000488281
Epoch 2425 | Loss: 239.4706268310547
Epoch 2450 | Loss: 197.9458770751953
Epoch 2475 | Loss: 449.9989013671875
Epoch 2500 | Loss: 348.60858154296875
Epoch 2525 | Loss: 503.93670654296875
Epoch 2550 | Loss: 2128.40283203125
Epoch 2575 | Loss: 409.5814208984375
Epoch 2600 | Loss: 1090.3759765625
Epoch 2625 | Loss: 209.4969482421875
Epoch 2650 | Loss: 702.425537109375
Epoch 2675 | Loss: 559.97412109375
Epoch 2700 | Loss: 994.8121948242188
Epoch 2725 | Loss: 408.59222412109375
Epoch 2750 | Loss: 406.1312255859375
Epoch 2775 | Loss: 335.3171691894531
Epoch 2800 | Loss: 539.6583862304688
Epoch 2825 | Loss: 416.9681396484375
Epoch 2850 | Loss: 677.041748046875
Epoch 2875 | Loss: 656.40380859375
Epoch 2900 | Loss: 792.5125122070312
Epoch 2925 | Loss: 429.66796875
Epoch 2950 | Loss: 616.0811767578125
Epoch 2975 | Loss: 969.8162841796875
Epoch 3000 | Loss: 2088.126220703125

**Comment by arimousa** [comment]
My interpretation of this behaviour in loss is the high stochasticity of diffusion models. One way to determine when to stop training is by measuring the reconstruction error, such as measuring RMSE on the training dataset.

**Comment by gustn9609** [comment]
Can I get a code that implements a reconstruction loss?
